app/jobs/generate_organization_report_job.rb

```diff
@@ -0,0 +1,30 @@
+class GenerateOrganizationReportJob < ApplicationJob
+  queue_as :default
+
+  def perform(id:, report_url:)
+    organization_report = OrganizationReport.find(id)
+    organization_report.update(status: "generating")
```

Comment:
May want a defensive check here to ensure it's not already generated or generating.

Not sure about the available statuses but something like this would work.

```ruby
organization_report = OrganizationReport.where(status: "new").find_by(id: id)
return unless organization_report
```